src/einsteinpy/symbolic/weyl.py

```diff
@@ -0,0 +1,82 @@
+import sympy
+
+from einsteinpy.symbolic.tensor import Tensor
+
+
+class WeylTensor(Tensor):
+    """
+    Class for defining Weyl Tensor
+    """
+
+    def __init__(self, arr, syms, t_riemann, t_ricci, r_scalar, t_metric):
+        """
+
+        Constructor and Initializer
+
+        Parameters
+        ----------
+        arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+            Sympy Array or multi-dimensional list containing Sympy Expressions
+        syms : tuple or list
+            Tuple of crucial symbols dentoting time-axis, 1st, 2nd, and 3rd axis (t,x1,x2,x3)
+        t_riemann : ~einsteinpy.symbolic.metric.RiemannCurvatureTensor
+            Riemann Tensor from which Weyl Tensor to be calculated
+        t_ricci : ~einsteinpy.symbolic.metric.RicciTensor
+            Ricci Tensor from which Weyl Tensor to be calculated
+        r_scalar : ~einsteinpy.symbolic.metric.RicciTensor
+            Ricci Scalar from which Weyl Tensor to be calculated
+        t_metric : ~einsteinpy.symbolic.metric.MetricTensor
+            Metric Tensor from which Weyl Tensor to be calculated
+
+        Raises
+        ------
+        TypeError
+            Raised when the dimension of the tensor is less than 3
+
+        """
+        super(WeylTensor, self).__init__(arr)
+        self.syms = syms
+        self.dims = len(self.syms)
+        if self.dims >= 3:
+            if self.dims == 3:
+                self.weyl = sympy.zeros(3, 3)
+
+            else:
+                self.weyl = self.calculate_weyl(t_riemann, t_ricci, r_scalar, t_metric)
+
+        else:
+            raise TypeError("Dimension should be 3 or more")
+
+    def calculate_weyl(self, t_riemann, t_ricci, r_scalar, t_metric):
```

Comment:
`calculate_weyl` not required!
Only a single classmethod `from_metric` is needed